### ISSUE TITLE
Support por python object CSV serialization in python 2

### ISSUE THREAD

**Issue by manugarri** [pydata/pandas#12959]
Let's say we have a file with 2 json encoded rows that we turn into a DataFrame:

```
{"first_name":"Manuel", "last_name":"Garrido Peña","experience":[{"name":"Peña Industries", "start":"May 2014"}]}
{"first_name":"Ana", "last_name":"Cañón", "experience":[{"name":"Cañón technologies","start":"April 2016"}]}
```

If one of the fields in a dataframe is a python object  ( e.g. a dictionary) with unicode strings on it, the 
output of `df.to_csv(NAME, encoding='utf8', index=False) is different with Python2 than with Python3.

```
#python2 output
experience,first_name,last_name
"[{u'start': u'May 2014', u'name': u'Peña Industries'}]",Manuel,Garrido Peña
"[{u'start': u'April 2016', u'name': u'Cañón technologies'}]",Ana,Cañón

#python3 output (expected)
experience,first_name,last_name
"[{'name': 'Peña Industries', 'start': 'May 2014'}]",Manuel,Garrido Peña
"[{'name': 'Cañón technologies', 'start': 'April 2016'}]",Ana,Cañón
```

Basically, it would make sense to be able to read back the `experience` field on that csv (maybe with a safe_eval?), and the unicode flags make it very hard to parse.

The files (and versions) for this example can be found on [this gist](https://gist.github.com/manugarri/4c0d68f9c2c8434654402e6053620b68)

**Comment by jreback** [comment]
pls have code that can be copy-pasted. include ``pd.show_versions()``. and I am not sure what you are asking.

**Comment by manugarri** [comment]
Sure, here is the python 2 `show_version` output.

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.19.0-58-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.17.1
nose: 1.3.7
pip: 8.0.3
setuptools: 20.1.1
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.17.0
statsmodels: None
IPython: 4.0.2
sphinx: None
patsy: None
dateutil: 2.4.2
pytz: 2015.7
blosc: 1.2.8
bottleneck: None
tables: 3.2.2
numexpr: 2.5.2
matplotlib: 1.5.1
openpyxl: None
xlrd: 0.9.4
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.4.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.5
pymysql: None
psycopg2: 2.6.1 (dt dec pq3 ext)
Jinja2: None
```

here is the python 3 `show_version` output: 

```
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.1.final.0
python-bits: 64
OS: Linux
OS-release: 3.19.0-58-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.0
nose: 1.3.7
pip: 8.1.1
setuptools: 20.2.2
Cython: 0.23.4
numpy: 1.11.0
scipy: 0.17.0
statsmodels: None
xarray: None
IPython: 4.1.1
sphinx: None
patsy: None
dateutil: 2.5.2
pytz: 2016.3
blosc: None
bottleneck: None
tables: None
numexpr: 2.4.6
matplotlib: 1.5.1
openpyxl: None
xlrd: 0.9.4
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: 2.6.1 (dt dec pq3 ext)
jinja2: 2.8
boto: 2.39.0
```
Given `test.jl` a file containing json encoded lines:
```
{"first_name":"Manuel", "last_name":"Garrido Peña","experience":[{"name":"Peña Industries", "start":"May 2014"}]}
{"first_name":"Ana", "last_name":"Cañón", "experience":[{"name":"Cañón technologies","start":"April 2016"}]}

```

in python 3:

```
with open('test.jl') as fname:
    lines = fname.readlines()
    df = pd.DataFrame([json.loads(l) for l in lines])
df.to_csv('test_p2.csv',index=False, encoding='utf8')
```

produces the expected output:

```
experience,first_name,last_name
"[{'name': 'Peña Industries', 'start': 'May 2014'}]",Manuel,Garrido Peña
"[{'name': 'Cañón technologies', 'start': 'April 2016'}]",Ana,Cañón
```

Same code, in python 2, produces the output:

```
experience,first_name,last_name
"[{u'start': u'May 2014', u'name': u'Peña Industries'}]",Manuel,Garrido Peña
"[{u'start': u'April 2016', u'name': u'Cañón technologies'}]",Ana,Cañón
```

with the `u` flag inside the string representation of the `experience` field in the csv file.

**Comment by jreback** [comment]
these are standard Unicode markers use read_json

http://pandas.pydata.org/pandas-docs/stable/io.html#io-json-reader
